pkgs/by-name/sh/sharkey/package.nix

```
@@ -0,0 +1,155 @@
+        nodejs
+        pnpm_9
+      ];
+      libPath = lib.makeLibraryPath [
+        ffmpeg-headless
+        jemalloc
+        stdenv.cc.cc
+      ];
+    in
+    ''
+      # adapted from repo dockerfile
+      # https://activitypub.software/TransFem-org/Sharkey/-/blob/develop/Dockerfile?ref_type=heads
+      mkdir -p $out/sharkey
+      cp -r built fluent-emojis node_modules package.json pnpm-workspace.yaml $out/sharkey/
+
+      mkdir -p $out/sharkey/packages/backend
+      cp -r packages/backend/{assets,built,migration,node_modules,ormconfig.js,package.json} $out/sharkey/packages/backend/
+      mkdir -p $out/sharkey/packages/backend/scripts
+      cp -r packages/backend/scripts/check_connect.js $out/sharkey/packages/backend/scripts/
+
+      mkdir -p $out/sharkey/packages/megalodon
+      cp -r packages/megalodon/{lib,node_modules,package.json} $out/sharkey/packages/megalodon/
+
+      mkdir -p $out/sharkey/packages/misskey-{bubble-game,js,reversi}
+      cp -r packages/misskey-bubble-game/{built,node_modules,package.json} $out/sharkey/packages/misskey-bubble-game/
+      cp -r packages/misskey-js/{built,node_modules,package.json} $out/sharkey/packages/misskey-js/
+      cp -r packages/misskey-reversi/{built,node_modules,package.json} $out/sharkey/packages/misskey-reversi/
+
+      mkdir -p $out/sharkey/packages/frontend{,-embed}
+      cp -r packages/frontend/assets $out/sharkey/packages/frontend/
+      cp -r packages/frontend-embed/assets $out/sharkey/packages/frontend-embed/
+
+      mkdir -p $out/sharkey/tossface-emojis
+      cp -r tossface-emojis/dist $out/sharkey/tossface-emojis/
+
+      # create a wrapper script for running sharkey commands (ie. alias for pnpm run)
+      makeWrapper ${lib.getExe pnpm_9} $out/bin/sharkey \
+        --chdir $out/sharkey \
+        --add-flags run \
+        --set-default NODE_ENV production \
+        --prefix PATH : ${binPath} \
+        --prefix LD_LIBRARY_PATH : ${libPath}
+    '';
+
+  fixupPhase = ''
```

You might want to add `runHook preFixup` and `runHook postFixup` to this phase (see: https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases). (Otherwise downstream overrides of these pre and post variants silently do nothing.)